src/app/encoder.cpp

```diff
@@ -2251,29 +2251,17 @@ PacketBufferHandle encodeGeneralCommissioningClusterCommissioningCompleteCommand
  * Command SetFabric
  */
 PacketBufferHandle encodeGeneralCommissioningClusterSetFabricCommand(uint8_t seqNum, EndpointId destinationEndpoint,
-                                                                     char * fabricId, char * fabricSecret, uint64_t breadcrumb,
-                                                                     uint32_t timeoutMs)
+                                                                     chip::BytesData fabricId, chip::BytesData fabricSecret,
+                                                                     uint64_t breadcrumb, uint32_t timeoutMs)
 {
     COMMAND_HEADER("SetFabric", GENERAL_COMMISSIONING_CLUSTER_ID);
-    size_t fabricIdStrLen = strlen(fabricId);
-    if (!CanCastTo<uint8_t>(fabricIdStrLen))
-    {
-        ChipLogError(Zcl, "Error encoding %s command. String too long: %d", kName, fabricIdStrLen);
-        return PacketBufferHandle();
-    }
-    size_t fabricSecretStrLen = strlen(fabricSecret);
-    if (!CanCastTo<uint8_t>(fabricSecretStrLen))
-    {
-        ChipLogError(Zcl, "Error encoding %s command. String too long: %d", kName, fabricSecretStrLen);
-        return PacketBufferHandle();
-    }
     buf.Put8(kFrameControlClusterSpecificCommand)
         .Put8(seqNum)
         .Put8(ZCL_SET_FABRIC_COMMAND_ID)
-        .Put(static_cast<uint8_t>(fabricIdStrLen))
-        .Put(fabricId)
-        .Put(static_cast<uint8_t>(fabricSecretStrLen))
-        .Put(fabricSecret)
+        .Put(static_cast<uint8_t>(fabricId.Length()))
```

Comment:
Why is this cast safe?  The code deleted above used to verify that the casts would be safe....